hrm-service/service-tests/contact-job/jobTypes/retrieve-transcript.test.ts

```diff
@@ -0,0 +1,337 @@
+import { isAfter } from 'date-fns';
+import each from 'jest-each';
+import timers from 'timers';
+
+import { withTaskId, accountSid, workerSid } from '../../mocks';
+import * as contactJobApi from '../../../src/contact-job/contact-job-data-access';
+import { db } from '../../../src/connection-pool';
+import '../../case-validation';
+import { CompletedContactJobBody } from '../../../src/contact-job/contact-job-messages';
+import { ContactMediaType } from '../../../src/contact/contact-json';
+import { chatChannels, Contact } from '../../../src/contact/contact';
+
+jest.mock('../../../src/contact-job/client-sns');
```

Comment:
As a note, I started playing with localstack and some other options for testing the lambda side of contact jobs yesterday. So there will likely be some examples of that soon that we could also use on the hrm-services side.

Reply:
The reason for this was just preventing the tests to actually dispatch anything. I didn't wanted to mock the sns library as that's an external dep, but instead I decided to mock `client-sns` which is a wrapper around it for our usecase. I could go a bit further and pass the "publish client function" as an argument for the publisher, which will make more of a black box I believe? Gonna do that if you think is worth.

Reply:
These are just comments on possible future direction, not asking for changes here (these tests are great btw - awesome to have these from the get go)

In the 'black box' approach we would actually dispatch something, but it would just go to an instance of AWS LocalStack we spun up for the test, rather than actual AWS.

For futher discussion anon :-)